<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Belinda Phipson" />

<meta name="date" content="2022-06-03" />

<title>Application to real single cell RNA-seq datasets</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">propeller-paper-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/phipsonlab/propeller-paper-analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Application to real single cell RNA-seq
datasets</h1>
<h4 class="author">Belinda Phipson</h4>
<h4 class="date">06/03/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-08-19
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>propeller-paper-analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220531code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220531)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220531code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220531)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomphipsonlabpropellerpaperanalysistree3bd56e03b4dbde80eb02bf8c23f6bd8720546984targetblank3bd56e0a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/phipsonlab/propeller-paper-analysis/tree/3bd56e03b4dbde80eb02bf8c23f6bd8720546984" target="_blank">3bd56e0</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomphipsonlabpropellerpaperanalysistree3bd56e03b4dbde80eb02bf8c23f6bd8720546984targetblank3bd56e0a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/phipsonlab/propeller-paper-analysis/tree/3bd56e03b4dbde80eb02bf8c23f6bd8720546984" target="_blank">3bd56e0</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/cold_warm_fresh_cellinfo.txt
    Ignored:    data/heartFYA.Rds
    Ignored:    data/pool_1.rds

Untracked files:
    Untracked:  analysis/PlotsForPaper.Rmd
    Untracked:  data/TypeIErrTables.Rdata
    Untracked:  data/appnote1cdata.rdata
    Untracked:  data/nullsimsVaryN_results.Rdata
    Untracked:  output/1x/
    Untracked:  output/Fig1ab.pdf
    Untracked:  output/Fig1cde.pdf
    Untracked:  output/Fig2ab.pdf
    Untracked:  output/Fig2abc.pdf
    Untracked:  output/Fig2c.pdf
    Untracked:  output/Figure1.pdf
    Untracked:  output/Figure1.png
    Untracked:  output/Figure2-01.png
    Untracked:  output/Figure2-with#.pdf
    Untracked:  output/Figure2-with#.png
    Untracked:  output/Figure2.ai
    Untracked:  output/Figure2.pdf
    Untracked:  output/Figure2.png
    Untracked:  output/Figure2_E_annotatedwithProp.pdf
    Untracked:  output/Figure3.pdf
    Untracked:  output/Figure3.png
    Untracked:  output/PDF/
    Untracked:  output/SuppFig4.pdf
    Untracked:  output/SuppFig4.png
    Untracked:  output/SuppTrueDiff10.pdf
    Untracked:  output/SuppTrueDiff20.pdf
    Untracked:  output/SuppTrueDiff3.pdf
    Untracked:  output/Supplementary Figure 2 v2.png
    Untracked:  output/Supplementary Figure 2.ai
    Untracked:  output/Supplementary Figure 2.pdf
    Untracked:  output/Supplementary Figure 2.png
    Untracked:  output/SupplementaryFigure3.pdf
    Untracked:  output/TrueDiffSimResults.Rda
    Untracked:  output/covidResults.pdf
    Untracked:  output/example_simdata.pdf
    Untracked:  output/extremeCaseTrueProps20CT.pdf
    Untracked:  output/fig2d.pdf
    Untracked:  output/gude-2022-06-27.log
    Untracked:  output/gude-2022-06-29.log
    Untracked:  output/heartResults.pdf
    Untracked:  output/heatmap20CT.pdf
    Untracked:  output/legend-fig2d.pdf
    Untracked:  output/pbmcOldYoungResults.pdf
    Untracked:  output/type1error5.csv
    Untracked:  output/typeIerrorResults.Rda

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/RealDataAnalysis.Rmd</code>) and
HTML (<code>docs/RealDataAnalysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/phipsonlab/propeller-paper-analysis/blob/3bd56e03b4dbde80eb02bf8c23f6bd8720546984/analysis/RealDataAnalysis.Rmd" target="_blank">3bd56e0</a>
</td>
<td>
bphipson
</td>
<td>
2022-08-19
</td>
<td>
update real data analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/phipsonlab/propeller-paper-analysis/f900e44ecf6150a2e055864fc495c522b107f8bd/docs/RealDataAnalysis.html" target="_blank">f900e44</a>
</td>
<td>
bphipson
</td>
<td>
2022-08-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/phipsonlab/propeller-paper-analysis/blob/258645364849f1d433de4d608c200cbe2899f63b/analysis/RealDataAnalysis.Rmd" target="_blank">2586453</a>
</td>
<td>
bphipson
</td>
<td>
2022-08-18
</td>
<td>
update analysis scripts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/phipsonlab/propeller-paper-analysis/1613612540c853767bd34935cc4f0293d44d3beb/docs/RealDataAnalysis.html" target="_blank">1613612</a>
</td>
<td>
bphipson
</td>
<td>
2022-06-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/phipsonlab/propeller-paper-analysis/blob/622390eb1559c52277dd15ea1b02566f63f4bac4/analysis/RealDataAnalysis.Rmd" target="_blank">622390e</a>
</td>
<td>
bphipson
</td>
<td>
2022-06-06
</td>
<td>
update real data analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/phipsonlab/propeller-paper-analysis/a39985a8740b5468e3b2d71315cf0c918dfa1505/docs/RealDataAnalysis.html" target="_blank">a39985a</a>
</td>
<td>
bphipson
</td>
<td>
2022-06-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/phipsonlab/propeller-paper-analysis/blob/cf6f450596772a8a0899b8542cc100899faa8f09/analysis/RealDataAnalysis.Rmd" target="_blank">cf6f450</a>
</td>
<td>
bphipson
</td>
<td>
2022-06-03
</td>
<td>
add real data analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>We analysed three different publicly available single cell datasets
to highlight the different types of models that can be fitted within the
propeller framework.</p>
<ul>
<li>Young and old female and male PBMCs
<ul>
<li>Huang, Zhaohao, Binyao Chen, Xiuxing Liu, He Li, Lihui Xie, Yuehan
Gao, Runping Duan, et al. 2021. “Effects of Sex and Aging on the Immune
Cell Landscape as Assessed by Single-Cell Transcriptomic Analysis.”
Proceedings of the National Academy of Sciences of the United States of
America 118 (33). <a href="https://doi.org/10.1073/pnas.2023216118"
class="uri">https://doi.org/10.1073/pnas.2023216118</a>.</li>
</ul></li>
<li>Healthy human heart biopsies across development
<ul>
<li>Sim, Choon Boon, Belinda Phipson, Mark Ziemann, Haloom Rafehi,
Richard J. Mills, Kevin I. Watt, Kwaku D. Abu-Bonsrah, et al. 2021.
“Sex-Specific Control of Human Heart Maturation by the Progesterone
Receptor.” Circulation 143 (16): 1614–28.</li>
</ul></li>
<li>Bronchoalveolar lavage fluid in a COVID19 dataset
<ul>
<li>Liao, Mingfeng, Yang Liu, Jing Yuan, Yanling Wen, Gang Xu, Juanjuan
Zhao, Lin Cheng, et al. 2020. “Single-Cell Landscape of Bronchoalveolar
Immune Cells in Patients with COVID-19.” Nature Medicine 26 (6):
842–44.</li>
</ul></li>
</ul>
<p>For all three datasets we use the logit transformation, except for
the covid dataset. In the covid dataset, there is clearly an outlier for
the plasma cell type. Using the logit transformation, plasma is found to
be significantly enriched in severe covid, but using the arcsin square
root transform shows that it is not significantly enriched. While logit
may be more powerful according to the simulations, it appears sensitive
to outliers, and in this case we would prefer to use the arcsin square
root transform.</p>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
<pre class="r"><code>library(speckle)
library(limma)
library(edgeR)
library(pheatmap)
library(gt)</code></pre>
<pre class="r"><code>source(&quot;./code/convertData.R&quot;)</code></pre>
</div>
<div id="young-and-old-female-and-male-pbmcs" class="section level1">
<h1>Young and old female and male PBMCs</h1>
<p>This dataset was published in PNAS in 2021 and examined PBMCs from 20
individuals. The dataset was balanced in terms of age and sex (5 samples
in each group: Male + Young, Male + Old, Female + Young, Female +
Old).</p>
<p>The data analysis reported in the paper was ANOVA directly on cell
type proportions modelling sex and age plus interaction. It was not
clear from the description whether separate models were fitted for sex
and age, removing the interaction term in order to interpret the effects
for sex and age correctly.</p>
<p>From the supplementary data I have extracted the cell type
proportions and using information from the paper I have converted the
proportions into the necessary data type (a list object) to analyse with
propeller.</p>
<div id="broad-cell-types-analysis" class="section level2">
<h2>Broad cell types analysis</h2>
<pre class="r"><code>sexCT &lt;- read.delim(&quot;./data/CTpropsTransposed.txt&quot;, row.names = 1)
sexprops &lt;- sexCT/100
prop.list &lt;- convertDataToList(sexprops,data.type=&quot;proportions&quot;, transform=&quot;logit&quot;,
                               scale.fac=174684/20)

sampleinfo &lt;- read.csv(&quot;./data/sampleinfo.csv&quot;, nrows = 20)
celltypes &lt;- read.csv(&quot;./data/CelltypeLevels.csv&quot;)
group.immune &lt;- paste(sampleinfo$Sex, sampleinfo$Age, sep=&quot;.&quot;)

celltypes$Celltype_L0 &lt;- celltypes$Celltype_L1
celltypes$Celltype_L0[celltypes$Celltype_L1 == &quot;CD4&quot; | celltypes$Celltype_L1 == &quot;CD8&quot; | celltypes$Celltype_L1 == &quot;CD4-CD8-&quot; | celltypes$Celltype_L1 == &quot;CD4+CD8+&quot; | celltypes$Celltype_L1 == &quot;T-mito&quot;] &lt;- &quot;TC&quot;</code></pre>
<pre class="r"><code>gt(data.frame(table(sampleinfo$Sex, sampleinfo$Age)), rownames_to_stub = TRUE, caption=&quot;Sample info for aging dataset&quot;)</code></pre>
<div id="aamalzggfc" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#aamalzggfc .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#aamalzggfc .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#aamalzggfc .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#aamalzggfc .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#aamalzggfc .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#aamalzggfc .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#aamalzggfc .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#aamalzggfc .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#aamalzggfc .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#aamalzggfc .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#aamalzggfc .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#aamalzggfc .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#aamalzggfc .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#aamalzggfc .gt_from_md > :first-child {
  margin-top: 0;
}

#aamalzggfc .gt_from_md > :last-child {
  margin-bottom: 0;
}

#aamalzggfc .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#aamalzggfc .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#aamalzggfc .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#aamalzggfc .gt_row_group_first td {
  border-top-width: 2px;
}

#aamalzggfc .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#aamalzggfc .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#aamalzggfc .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#aamalzggfc .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#aamalzggfc .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#aamalzggfc .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#aamalzggfc .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#aamalzggfc .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#aamalzggfc .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#aamalzggfc .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-left: 4px;
  padding-right: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#aamalzggfc .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#aamalzggfc .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#aamalzggfc .gt_left {
  text-align: left;
}

#aamalzggfc .gt_center {
  text-align: center;
}

#aamalzggfc .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#aamalzggfc .gt_font_normal {
  font-weight: normal;
}

#aamalzggfc .gt_font_bold {
  font-weight: bold;
}

#aamalzggfc .gt_font_italic {
  font-style: italic;
}

#aamalzggfc .gt_super {
  font-size: 65%;
}

#aamalzggfc .gt_two_val_uncert {
  display: inline-block;
  line-height: 1em;
  text-align: right;
  font-size: 60%;
  vertical-align: -0.25em;
  margin-left: 0.1em;
}

#aamalzggfc .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 75%;
  vertical-align: 0.4em;
}

#aamalzggfc .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#aamalzggfc .gt_slash_mark {
  font-size: 0.7em;
  line-height: 0.7em;
  vertical-align: 0.15em;
}

#aamalzggfc .gt_fraction_numerator {
  font-size: 0.6em;
  line-height: 0.6em;
  vertical-align: 0.45em;
}

#aamalzggfc .gt_fraction_denominator {
  font-size: 0.6em;
  line-height: 0.6em;
  vertical-align: -0.05em;
}
</style>
<table class="gt_table">
  <caption>Sample info for aging dataset</caption>
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1"></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">Var1</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">Var2</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">Freq</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td class="gt_row gt_right gt_stub">1</td>
<td class="gt_row gt_center">female</td>
<td class="gt_row gt_center">old</td>
<td class="gt_row gt_right">5</td></tr>
    <tr><td class="gt_row gt_right gt_stub">2</td>
<td class="gt_row gt_center">male</td>
<td class="gt_row gt_center">old</td>
<td class="gt_row gt_right">5</td></tr>
    <tr><td class="gt_row gt_right gt_stub">3</td>
<td class="gt_row gt_center">female</td>
<td class="gt_row gt_center">young</td>
<td class="gt_row gt_right">5</td></tr>
    <tr><td class="gt_row gt_right gt_stub">4</td>
<td class="gt_row gt_center">male</td>
<td class="gt_row gt_center">young</td>
<td class="gt_row gt_right">5</td></tr>
  </tbody>
  
  
</table>
</div>
<pre class="r"><code>levels(factor(celltypes$Celltype_L0))</code></pre>
<pre><code>[1] &quot;BC&quot; &quot;DC&quot; &quot;MC&quot; &quot;NK&quot; &quot;TC&quot;</code></pre>
<pre class="r"><code>sexprops.broad &lt;-  matrix(NA,nrow=length(levels(factor(celltypes$Celltype_L0))), ncol=ncol(sexprops)) 
rownames(sexprops.broad) &lt;- levels(factor(celltypes$Celltype_L0))
colnames(sexprops.broad) &lt;- colnames(sexprops)

for(i in 1:ncol(sexprops.broad)) sexprops.broad[,i] &lt;- tapply(sexprops[,i],celltypes$Celltype_L0,sum)

prop.broad.list &lt;- convertDataToList(sexprops.broad,data.type=&quot;proportions&quot;, transform=&quot;logit&quot;, scale.fac=174684/20)</code></pre>
<pre class="r"><code>par(mfrow=c(1,5))
par(mar=c(6,5,3,2))
for(i in 1:nrow(sexprops.broad)){
  stripchart(as.numeric(sexprops.broad[i,])~group.immune,
           vertical=TRUE, pch=c(8,16), method=&quot;jitter&quot;,
           col = c(ggplotColors(20)[20],&quot;hotpink&quot;,4, &quot;darkblue&quot;),cex=2,
           ylab=&quot;Proportions&quot;, cex.axis=1.25, cex.lab=1.5,
           group.names=c(&quot;F.Old&quot;,&quot;F.Young&quot;,&quot;M.Old&quot;,&quot;M.Young&quot;), las=2)
  title(rownames(sexprops.broad)[i], cex.main=1.5, adj=0)
}</code></pre>
<p><img src="figure/RealDataAnalysis.Rmd/unnamed-chunk-6-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>designAS &lt;- model.matrix(~0+sampleinfo$Age + sampleinfo$Sex)
colnames(designAS) &lt;- c(&quot;old&quot;,&quot;young&quot;,&quot;MvsF&quot;)

# Young vs old
mycontr &lt;- makeContrasts(young-old, levels=designAS)
propeller.ttest(prop.list = prop.broad.list,design = designAS, contrasts = mycontr,
                robust=TRUE,trend=FALSE,sort=TRUE)</code></pre>
<pre><code>   PropMean.old PropMean.young PropRatio Tstatistic    P.Value        FDR
MC   0.19656790     0.14320118 0.7285074 -2.4782763 0.01517944 0.07589718
TC   0.53445740     0.60263724 1.1275684  1.7958220 0.07607623 0.19019057
NK   0.15808995     0.13347185 0.8442779 -1.2139485 0.22812958 0.38021597
BC   0.09432895     0.10249865 1.0866086  0.8453907 0.40026618 0.50033273
DC   0.01655580     0.01819108 1.0987734  0.2439462 0.80786033 0.80786033</code></pre>
<pre class="r"><code>designSex &lt;- model.matrix(~0+sampleinfo$Sex + sampleinfo$Age)
colnames(designSex) &lt;- c(&quot;female&quot;,&quot;male&quot;,&quot;YvO&quot;)

# Male vs female
mycontr &lt;- makeContrasts(male-female, levels=designSex)
propeller.ttest(prop.list = prop.broad.list,design = designSex, contrasts = mycontr,
                robust=TRUE,trend=FALSE,sort=TRUE)</code></pre>
<pre><code>   PropMean.female PropMean.male PropRatio  Tstatistic     P.Value       FDR
NK       0.1169176    0.17464421 1.4937377  2.68780027 0.008652239 0.0432612
TC       0.5973764    0.53971825 0.9034811 -1.50690429 0.135542825 0.3388571
DC       0.0157718    0.01897507 1.2031011  1.09443016 0.276858018 0.4614300
BC       0.1011660    0.09566164 0.9455912 -0.46829428 0.640772810 0.8009660
MC       0.1687683    0.17100082 1.0132286  0.06740726 0.946415797 0.9464158</code></pre>
</div>
<div id="refined-cell-types-analysis" class="section level2">
<h2>Refined cell types analysis</h2>
</div>
<div id="young-vs-old" class="section level2">
<h2>Young vs Old</h2>
<p>Set up design matrix using a means model, taking into account age and
sex.</p>
<pre class="r"><code>designAS &lt;- model.matrix(~0+sampleinfo$Age + sampleinfo$Sex)
colnames(designAS) &lt;- c(&quot;old&quot;,&quot;young&quot;,&quot;MvsF&quot;)

# Young vs old
mycontr &lt;- makeContrasts(young-old, levels=designAS)
propeller.ttest(prop.list = prop.list,design = designAS, contrasts = mycontr,
                robust=TRUE,trend=FALSE,sort=TRUE)</code></pre>
<pre><code>          PropMean.old PropMean.young PropRatio  Tstatistic      P.Value
CD8.Naive 0.0251316033   0.0901386289 3.5866645  4.56531629 0.0001655915
CD16      0.0346697484   0.0184110417 0.5310405 -3.53845105 0.0019313526
T-mito    0.0050757865   0.0030999505 0.6107330 -2.70564864 0.0130970606
INTER     0.0237901860   0.0160354568 0.6740366 -2.34264441 0.0288651941
CD4-CD8-  0.0160339954   0.0301971880 1.8833227  2.26893189 0.0338557770
TREG      0.0224289338   0.0173984106 0.7757128 -2.26167176 0.0342259694
CD14      0.1381079661   0.1087546814 0.7874613 -1.75042587 0.0943497196
ABC       0.0042268748   0.0024242272 0.5735271 -1.53059704 0.1406904913
CD4.Naive 0.1048008357   0.1320789977 1.2602857  1.46557604 0.1574740245
MBC       0.0358736549   0.0452983191 1.2627183  1.33884893 0.1948341805
CD8.CTL   0.1154952554   0.0782456785 0.6774796 -1.31085637 0.2039626408
PC        0.0054708831   0.0049397487 0.9029162 -1.26307895 0.2203089287
pre-DC    0.0002285096   0.0003193005 1.3973177  1.24144109 0.2280358887
CDC1      0.0005479607   0.0008647764 1.5781725  1.19434388 0.2455671742
CD8.Tem   0.0639345157   0.0802785262 1.2556367  1.14556222 0.2646082927
NK3       0.0886767142   0.0704022112 0.7939199 -1.02022467 0.3191513408
NK1       0.0125858088   0.0102540186 0.8147286 -0.64532282 0.5256577519
NBC       0.0487575349   0.0498363513 1.0221261  0.49890406 0.6229966249
CD4.Tem   0.0873700973   0.0832684783 0.9530547 -0.40297138 0.6909611645
CD4.Tcm   0.0799349205   0.0746146497 0.9334425  0.23348683 0.8176285211
PDC       0.0031666984   0.0034597825 1.0925519  0.20366298 0.8405637766
CD4+CD8+  0.0142514516   0.0133167354 0.9344126 -0.11420442 0.9101389532
NK2       0.0568274310   0.0528156246 0.9294037  0.04492635 0.9645875142
CDC2      0.0126126342   0.0135472159 1.0740989  0.03134858 0.9752816776
                  FDR
CD8.Naive 0.003974195
CD16      0.023176231
T-mito    0.104776485
INTER     0.136903877
CD4-CD8-  0.136903877
TREG      0.136903877
CD14      0.323484753
ABC       0.419930732
CD4.Naive 0.419930732
MBC       0.420972299
CD8.CTL   0.420972299
PC        0.420972299
pre-DC    0.420972299
CDC1      0.420972299
CD8.Tem   0.423373268
NK3       0.478727011
NK1       0.742105062
NBC       0.830662167
CD4.Tem   0.872793050
CD4.Tcm   0.960644316
PDC       0.960644316
CD4+CD8+  0.975281678
NK2       0.975281678
CDC2      0.975281678</code></pre>
<p>Visualise significant cell types:</p>
<pre class="r"><code>par(mfrow=c(1,2))
stripchart(as.numeric(sexprops[&quot;CD8.Naive&quot;,])~group.immune,
           vertical=TRUE, pch=c(8,16), method=&quot;jitter&quot;,
           col = c(ggplotColors(20)[20],&quot;hotpink&quot;,4, &quot;darkblue&quot;),cex=2,
           ylab=&quot;Proportions&quot;, cex.axis=1.25, cex.lab=1.5,
           group.names=c(&quot;F.Old&quot;,&quot;F.Young&quot;,&quot;M.Old&quot;,&quot;M.Young&quot;))
title(&quot;CD8.Naive: Young Vs Old&quot;, cex.main=1.5, adj=0)
text(3.2,0.18, labels = &quot;Adj.Pval = 0.004&quot;)

stripchart(as.numeric(sexprops[&quot;CD16&quot;,])~group.immune,
           vertical=TRUE, pch=c(8,16), method=&quot;jitter&quot;,
           col = c(ggplotColors(20)[20],&quot;hotpink&quot;,4, &quot;darkblue&quot;),cex=2,
           ylab=&quot;Proportions&quot;, cex.axis=1.25, cex.lab=1.5,
           group.names=c(&quot;F.Old&quot;,&quot;F.Young&quot;,&quot;M.Old&quot;,&quot;M.Young&quot;))
title(&quot;CD16: Young Vs Old&quot;, cex.main=1.5, adj=0)
text(2.2,0.049, labels = &quot;Adj.Pval = 0.023&quot;)</code></pre>
<p><img src="figure/RealDataAnalysis.Rmd/unnamed-chunk-10-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pbmc.oldyoung &lt;- propeller.ttest(prop.list = prop.list,design = designAS, contrasts = mycontr,
                robust=TRUE,trend=FALSE,sort=TRUE)
sig.pbmc &lt;- rownames(pbmc.oldyoung)

pdf(file=&quot;output/pbmcOldYoungResults.pdf&quot;,width = 13, height=13)
par(mfrow=c(6,4))
par(mar=c(4,5,2,2))
for(i in 1:length(sig.pbmc)){
stripchart(as.numeric(sexprops[sig.pbmc[i],])~group.immune,
           vertical=TRUE, pch=c(8,16), method=&quot;jitter&quot;,
           col = c(ggplotColors(20)[20],&quot;hotpink&quot;,4, &quot;darkblue&quot;),cex=2,
           ylab=&quot;Proportions&quot;, cex.axis=1.25, cex.lab=1.5,
           group.names=c(&quot;F.Old&quot;,&quot;F.Young&quot;,&quot;M.Old&quot;,&quot;M.Young&quot;))
title(paste(sig.pbmc[i],&quot;: Young Vs Old&quot;, sep=&quot;&quot;), cex.main=1.5, adj=0)
legend(&quot;top&quot;, legend = paste(&quot;Adj.Pval = &quot;,round(pbmc.oldyoung$FDR,3)[i],sep=&quot;&quot;),cex=1.5,bty=&quot;n&quot;,bg=&quot;n&quot;)  
}
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
<div id="male-vs-female" class="section level2">
<h2>Male vs female</h2>
<pre class="r"><code>designSex &lt;- model.matrix(~0+sampleinfo$Sex + sampleinfo$Age)
colnames(designSex) &lt;- c(&quot;female&quot;,&quot;male&quot;,&quot;YvO&quot;)

# Male vs female
mycontr &lt;- makeContrasts(male-female, levels=designSex)
propeller.ttest(prop.list = prop.list,design = designSex, contrasts = mycontr,
                robust=TRUE,trend=FALSE,sort=TRUE)</code></pre>
<pre><code>          PropMean.female PropMean.male PropRatio Tstatistic     P.Value
PC           0.0071388502   0.003271782 0.4583065 -3.1669247 0.004618248
pre-DC       0.0001775081   0.000370302 2.0861130  2.5110134 0.020226075
CDC2         0.0114064513   0.014753399 1.2934258  1.6325785 0.117165640
NK2          0.0457752659   0.063867790 1.3952467  1.5116244 0.145428553
CD4-CD8-     0.0169315724   0.029299611 1.7304720  1.5102803 0.145769120
T-mito       0.0034714265   0.004704310 1.3551520  1.4058711 0.174095994
NK3          0.0607526719   0.098326253 1.6184680  1.1335501 0.269669984
NK1          0.0103896562   0.012450171 1.1983237  1.1164933 0.276735756
CD8.CTL      0.1180843551   0.075656579 0.6406994 -1.1131493 0.278136717
CDC1         0.0005883982   0.000824339 1.4009883  1.0905509 0.287740106
PDC          0.0035994459   0.003027035 0.8409725 -1.0645655 0.299076537
CD8.Naive    0.0547935108   0.060476721 1.1037205  0.9733622 0.341375491
CD8.Tem      0.0747782768   0.069434765 0.9285419 -0.7670999 0.451391030
TREG         0.0209130057   0.018914339 0.9044295 -0.7245891 0.476541303
CD16         0.0281162892   0.024964501 0.8879017 -0.7035509 0.489386405
ABC          0.0035141234   0.003136979 0.8926774 -0.5597871 0.581505744
CD4.Naive    0.1243810258   0.112498808 0.9044692 -0.5287280 0.602499455
NBC          0.0497589512   0.048834935 0.9814301  0.4375921 0.666118271
CD4+CD8+     0.0134400072   0.014128180 1.0512033  0.4252468 0.674896446
MBC          0.0407540321   0.040417942 0.9917532 -0.3094240 0.760026981
INTER        0.0206287843   0.019196858 0.9305860 -0.2751525 0.785834451
CD14         0.1200231846   0.126839463 1.0567913  0.2749839 0.785962262
CD4.Tcm      0.0843546548   0.070194915 0.8321404 -0.1934495 0.848452765
CD4.Tem      0.0862285523   0.084410023 0.9789104 -0.1642394 0.871081911
                FDR
PC        0.1108380
pre-DC    0.2427129
CDC2      0.6525306
NK2       0.6525306
CD4-CD8-  0.6525306
T-mito    0.6525306
NK3       0.6525306
NK1       0.6525306
CD8.CTL   0.6525306
CDC1      0.6525306
PDC       0.6525306
CD8.Naive 0.6827510
CD8.Tem   0.7830182
TREG      0.7830182
CD16      0.7830182
ABC       0.8505875
CD4.Naive 0.8505875
NBC       0.8525008
CD4+CD8+  0.8525008
MBC       0.8574134
INTER     0.8574134
CD14      0.8574134
CD4.Tcm   0.8710819
CD4.Tem   0.8710819</code></pre>
</div>
</div>
<div id="heart-development-analysis" class="section level1">
<h1>Heart development analysis</h1>
<p>This dataset was published by Sim et al in 2021 and looked at heart
development across fetal, young and adult developmental time points.
There are three samples at each developmental time point. There is also
mix of male and female samples and one of the key findings of the study
was transcriptional differences in cardiomyocyte development between
males and females.</p>
<p>Here we look at differences in cellular composition of human hearts
across the developmental trajectory, taking sex into account.</p>
<p>We can simply perform an anova test to find any differences between
the three time points. The propeller.anova function can be called to do
this directly.</p>
<p>We can also look at development as a continuous trajectory and model
development as a continuous variable by getting the transformed
proportions and using fitting functions in limma directly.</p>
<p>Both of these analysis approaches are shown below.</p>
<pre class="r"><code>heart.info &lt;- read.csv(file=&quot;./data/cellinfo.csv&quot;, row.names = 1)
heart.counts &lt;- table(heart.info$Celltype, heart.info$Sample)
trueprops &lt;- rowSums(heart.counts)/sum(rowSums(heart.counts))

heart.info$Group &lt;- NA
heart.info$Group[grep(&quot;f&quot;,heart.info$Sample)] &lt;- &quot;fetal&quot;
heart.info$Group[grep(&quot;y&quot;,heart.info$Sample)] &lt;- &quot;young&quot;
heart.info$Group[grep(&quot;a&quot;,heart.info$Sample)] &lt;- &quot;adult&quot;

sample &lt;- factor(heart.info$Sample, levels= paste(rep(c(&quot;f&quot;,&quot;y&quot;,&quot;a&quot;), each=3),c(1:3),sep=&quot;&quot;))
group &lt;- factor(heart.info$Group, levels=c(&quot;fetal&quot;,&quot;young&quot;,&quot;adult&quot;))</code></pre>
<pre class="r"><code>grp &lt;- factor(rep(c(&quot;fetal&quot;,&quot;young&quot;,&quot;adult&quot;),each=3), levels=c(&quot;fetal&quot;,&quot;young&quot;,&quot;adult&quot;))
sex &lt;- factor(c(&quot;m&quot;,&quot;m&quot;,&quot;f&quot;,&quot;m&quot;,&quot;f&quot;,&quot;m&quot;,&quot;f&quot;,&quot;m&quot;,&quot;m&quot;))
dose &lt;- rep(c(1,2,3), each=3) </code></pre>
<p>The sample information is summarised below:</p>
<pre class="r"><code>gt(data.frame(Sample=1:9,Group=grp, Sex=sex),caption=&quot;Sample info for heart dataset&quot;)</code></pre>
<div id="abgrewqzol" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#abgrewqzol .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#abgrewqzol .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#abgrewqzol .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#abgrewqzol .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#abgrewqzol .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#abgrewqzol .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#abgrewqzol .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#abgrewqzol .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#abgrewqzol .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#abgrewqzol .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#abgrewqzol .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#abgrewqzol .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#abgrewqzol .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#abgrewqzol .gt_from_md > :first-child {
  margin-top: 0;
}

#abgrewqzol .gt_from_md > :last-child {
  margin-bottom: 0;
}

#abgrewqzol .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#abgrewqzol .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#abgrewqzol .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#abgrewqzol .gt_row_group_first td {
  border-top-width: 2px;
}

#abgrewqzol .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#abgrewqzol .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#abgrewqzol .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#abgrewqzol .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#abgrewqzol .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#abgrewqzol .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#abgrewqzol .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#abgrewqzol .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#abgrewqzol .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#abgrewqzol .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-left: 4px;
  padding-right: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#abgrewqzol .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#abgrewqzol .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#abgrewqzol .gt_left {
  text-align: left;
}

#abgrewqzol .gt_center {
  text-align: center;
}

#abgrewqzol .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#abgrewqzol .gt_font_normal {
  font-weight: normal;
}

#abgrewqzol .gt_font_bold {
  font-weight: bold;
}

#abgrewqzol .gt_font_italic {
  font-style: italic;
}

#abgrewqzol .gt_super {
  font-size: 65%;
}

#abgrewqzol .gt_two_val_uncert {
  display: inline-block;
  line-height: 1em;
  text-align: right;
  font-size: 60%;
  vertical-align: -0.25em;
  margin-left: 0.1em;
}

#abgrewqzol .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 75%;
  vertical-align: 0.4em;
}

#abgrewqzol .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#abgrewqzol .gt_slash_mark {
  font-size: 0.7em;
  line-height: 0.7em;
  vertical-align: 0.15em;
}

#abgrewqzol .gt_fraction_numerator {
  font-size: 0.6em;
  line-height: 0.6em;
  vertical-align: 0.45em;
}

#abgrewqzol .gt_fraction_denominator {
  font-size: 0.6em;
  line-height: 0.6em;
  vertical-align: -0.05em;
}
</style>
<table class="gt_table">
  <caption>Sample info for heart dataset</caption>
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">Sample</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">Group</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">Sex</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td class="gt_row gt_right">1</td>
<td class="gt_row gt_center">fetal</td>
<td class="gt_row gt_center">m</td></tr>
    <tr><td class="gt_row gt_right">2</td>
<td class="gt_row gt_center">fetal</td>
<td class="gt_row gt_center">m</td></tr>
    <tr><td class="gt_row gt_right">3</td>
<td class="gt_row gt_center">fetal</td>
<td class="gt_row gt_center">f</td></tr>
    <tr><td class="gt_row gt_right">4</td>
<td class="gt_row gt_center">young</td>
<td class="gt_row gt_center">m</td></tr>
    <tr><td class="gt_row gt_right">5</td>
<td class="gt_row gt_center">young</td>
<td class="gt_row gt_center">f</td></tr>
    <tr><td class="gt_row gt_right">6</td>
<td class="gt_row gt_center">young</td>
<td class="gt_row gt_center">m</td></tr>
    <tr><td class="gt_row gt_right">7</td>
<td class="gt_row gt_center">adult</td>
<td class="gt_row gt_center">f</td></tr>
    <tr><td class="gt_row gt_right">8</td>
<td class="gt_row gt_center">adult</td>
<td class="gt_row gt_center">m</td></tr>
    <tr><td class="gt_row gt_right">9</td>
<td class="gt_row gt_center">adult</td>
<td class="gt_row gt_center">m</td></tr>
  </tbody>
  
  
</table>
</div>
<div id="anova-test-with-propeller.anova" class="section level2">
<h2>Anova test with propeller.anova</h2>
<p>For the original analysis in the published paper by Sim et al., we
performed an ANOVA using propeller(logit).</p>
<pre class="r"><code>prop.logit &lt;- getTransformedProps(clusters = heart.info$Celltype, sample=sample,
                                  transform = &quot;logit&quot;)
design.anova &lt;- model.matrix(~0+grp+sex)

propeller.anova(prop.logit,design = design.anova, coef=c(1,2,3), robust=TRUE, 
                trend = FALSE, sort=TRUE)</code></pre>
<pre><code>                    PropMean.grpfetal PropMean.grpyoung PropMean.grpadult
Erythroid                 0.004433044        0.00000000       0.000000000
Immune cells              0.027545963        0.10875124       0.189587828
Cardiomyocytes            0.682410381        0.42676145       0.273546585
Fibroblast                0.111342233        0.26192406       0.298689329
Neurons                   0.012643346        0.02620977       0.011380837
Epicardial cells          0.051414853        0.07541028       0.093157709
Smooth muscle cells       0.008101973        0.00846540       0.009099294
Endothelial cells         0.102108207        0.09247781       0.124538418
                    Fstatistic      P.Value          FDR
Erythroid           42.8302506 1.162495e-13 9.299960e-13
Immune cells        10.3529005 9.295704e-05 3.718282e-04
Cardiomyocytes       7.6881182 8.440468e-04 2.250792e-03
Fibroblast           4.0630697 2.057471e-02 4.114942e-02
Neurons              1.3712270 2.592465e-01 4.147943e-01
Epicardial cells     0.7965363 4.541612e-01 6.055483e-01
Smooth muscle cells  0.3492357 7.062149e-01 7.326515e-01
Endothelial cells    0.3122046 7.326515e-01 7.326515e-01</code></pre>
</div>
<div id="modelling-development-as-a-continuous-variable"
class="section level2">
<h2>Modelling development as a continuous variable</h2>
<p>Here we model the data in a different way with development as a
continuous variable, and include sex as an additional covariate to
control for.</p>
<pre class="r"><code>des.dose &lt;- model.matrix(~dose + sex)
des.dose</code></pre>
<pre><code>  (Intercept) dose sexm
1           1    1    1
2           1    1    1
3           1    1    0
4           1    2    1
5           1    2    0
6           1    2    1
7           1    3    0
8           1    3    1
9           1    3    1
attr(,&quot;assign&quot;)
[1] 0 1 2
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$sex
[1] &quot;contr.treatment&quot;</code></pre>
<pre class="r"><code>fit &lt;- lmFit(prop.logit$TransformedProps,des.dose)
fit &lt;- eBayes(fit, robust=TRUE)
topTable(fit,coef=2)</code></pre>
<pre><code>                          logFC    AveExpr          t      P.Value   adj.P.Val
Immune cells         1.01359669 -2.4222033  4.2685426 0.0007551546 0.004980697
Erythroid           -1.62960543 -7.8002460 -4.0140721 0.0012451742 0.004980697
Cardiomyocytes      -0.91607837 -0.1936611 -3.4462160 0.0038586975 0.010289860
Fibroblast           0.62700537 -1.3756904  2.5326823 0.0236986525 0.047397305
Epicardial cells     0.29219063 -2.6566961  1.2026143 0.2487822518 0.398051603
Smooth muscle cells  0.18767427 -4.9125588  0.6717760 0.5125008180 0.603572791
Endothelial cells    0.13688475 -2.1652913  0.6467132 0.5281261918 0.603572791
Neurons             -0.05626966 -4.1954452 -0.2315895 0.8201585898 0.820158590
                             B
Immune cells        -0.3850285
Erythroid           -0.8666233
Cardiomyocytes      -1.9515138
Fibroblast          -3.6607326
Epicardial cells    -5.6855671
Smooth muscle cells -6.1705968
Endothelial cells   -6.1872807
Neurons             -6.3739773</code></pre>
<pre class="r"><code>fit.plot &lt;- lmFit(prop.logit$Proportions,des.dose)
fit.plot &lt;- eBayes(fit.plot, robust=TRUE)</code></pre>
<p>The two analyses identify the same cell types as significantly
enriched/depleted, although there is a change in the order of
significance.</p>
<p>Three significant cell types are visualised below.</p>
<pre class="r"><code>par(mfrow=c(1,3))
stripchart(as.numeric(prop.logit$Proportions[&quot;Immune cells&quot;,])~grp,
           vertical=TRUE, pch=16, method=&quot;jitter&quot;,
           col = ggplotColors(4),cex=2, 
           ylab=&quot;Proportions&quot;,cex.axis=1.25, cex.lab=1.5)
title(&quot;Immune development&quot;, cex.main=1.5, adj=0)
abline(a=fit.plot$coefficients[&quot;Immune cells&quot;,1], b=fit.plot$coefficients[&quot;Immune cells&quot;,2], lty=2, lwd=2)


stripchart(as.numeric(prop.logit$Proportions[&quot;Cardiomyocytes&quot;,])~grp,
           vertical=TRUE, pch=16, method=&quot;jitter&quot;,
           col = ggplotColors(4),cex=2, 
           ylab=&quot;Proportions&quot;,cex.axis=1.25, cex.lab=1.5)
title(&quot;Cardiomyocyte development&quot;, cex.main=1.5, adj=0)
abline(a=fit.plot$coefficients[&quot;Cardiomyocytes&quot;,1], b=fit.plot$coefficients[&quot;Cardiomyocytes&quot;,2], lty=2, lwd=2)
text(2.6,0.77, labels = &quot;Adj.Pval = 0.01&quot;)

stripchart(as.numeric(prop.logit$Proportions[&quot;Fibroblast&quot;,])~grp,
           vertical=TRUE, pch=16, method=&quot;jitter&quot;,
           col = ggplotColors(4),cex=2, 
           ylab=&quot;Proportions&quot;,cex.axis=1.25, cex.lab=1.5)
title(&quot;Fibroblast development&quot;, cex.main=1.5, adj=0)
abline(a=fit.plot$coefficients[&quot;Fibroblast&quot;,1], b=fit.plot$coefficients[&quot;Fibroblast&quot;,2], lty=2, lwd=2)</code></pre>
<p><img src="figure/RealDataAnalysis.Rmd/unnamed-chunk-18-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sig.heart &lt;- rownames(topTable(fit,coef=2))
grp &lt;- factor(rep(c(&quot;fetal&quot;,&quot;young&quot;,&quot;adult&quot;),each=3), levels=c(&quot;fetal&quot;,&quot;young&quot;,&quot;adult&quot;))

pdf(file=&quot;output/heartResults.pdf&quot;,width = 13, height=6)
par(mfrow=c(2,4))
par(mar=c(4,5,2,2))
for(i in 1:length(sig.heart)){
  
stripchart(as.numeric(prop.logit$Proportions[sig.heart[i],])~grp,
           vertical=TRUE, pch=16, method=&quot;jitter&quot;,
           col = ggplotColors(4),cex=2, 
           ylab=&quot;Proportions&quot;,cex.axis=1.25, cex.lab=1.5)
title(sig.heart[i], cex.main=1.5, adj=0)
legend(&quot;top&quot;, legend = paste(&quot;Adj.Pval = &quot;,round(topTable(fit,coef=2)$adj.P.Val,3)[i],sep=&quot;&quot;),cex=1.5,bty=&quot;n&quot;,bg=&quot;n&quot;)
#abline(a=fit.plot$coefficients[sig.heart[i],1], b=fit.plot$coefficients[sig.heart[i],2], lty=2, lwd=2)
}
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
</div>
<div id="covid-data" class="section level1">
<h1>COVID data</h1>
<p>This dataset was published by Liao et al in 2020 in Nature Medicine.
They compared moderate and severe covid to healthy controls. They
sampled bronchoalveolar lavage fluid from each individual.</p>
<p>For this dataset there are no additional covariates and we use the
propeller() function with the cell level annotation information for cell
type, sample, and group. The function automatically detects more than
two groups and performs an ANOVA for each cell type. We fit with both
the logit and arcsin square root transformed data to show the effect of
an outlier sample in the plasma cell type.</p>
<pre class="r"><code>covid &lt;- read.delim(&quot;./data/covid.cell.annotation.meta.txt&quot;)</code></pre>
<p>propeller using logit transformed proportions:</p>
<pre class="r"><code>output.logit &lt;- propeller(clusters=covid$celltype, sample=covid$sample_new, group=covid$group, transform=&quot;logit&quot;)
output.logit</code></pre>
<pre><code>            BaselineProp  PropMean.HC  PropMean.M  PropMean.S Fstatistic
Neutrophil   0.024417668 0.0000000000 0.001204431 0.055593846 34.5347175
Plasma       0.015817544 0.0002244276 0.002149909 0.050912723  8.7181367
pDC          0.002309574 0.0011342772 0.008209773 0.001065692  5.7852728
NK           0.016425326 0.0088943293 0.052465923 0.017978750  5.3912409
T            0.117241275 0.0945944806 0.325029563 0.137097435  3.1551241
mDC          0.014860286 0.0237761816 0.030831149 0.008875553  2.4842289
B            0.003342805 0.0035032662 0.012989518 0.004005391  2.3945864
Epithelial   0.053652014 0.1302459194 0.051903406 0.118455372  1.8134669
Macrophages  0.750869889 0.7352897831 0.512995844 0.604316237  1.6207397
Mast         0.001063620 0.0023373349 0.002220485 0.001699001  0.6920735
                 P.Value          FDR
Neutrophil  3.546468e-07 3.546468e-06
Plasma      2.056673e-03 1.028336e-02
pDC         1.052218e-02 3.493472e-02
NK          1.397389e-02 3.493472e-02
T           6.468749e-02 1.293750e-01
mDC         1.090338e-01 1.673738e-01
B           1.171617e-01 1.673738e-01
Epithelial  1.901866e-01 2.377332e-01
Macrophages 2.239111e-01 2.487901e-01
Mast        5.127128e-01 5.127128e-01</code></pre>
<p>propeller using arcsin square root transformed proportions:</p>
<pre class="r"><code>output.asin &lt;- propeller(clusters=covid$celltype, sample=covid$sample_new, group=covid$group, transform=&quot;asin&quot;)
output.asin</code></pre>
<pre><code>            BaselineProp  PropMean.HC  PropMean.M  PropMean.S Fstatistic
pDC          0.002309574 0.0011342772 0.008209773 0.001065692 7.66800413
Neutrophil   0.024417668 0.0000000000 0.001204431 0.055593846 7.69977312
NK           0.016425326 0.0088943293 0.052465923 0.017978750 7.52069109
T            0.117241275 0.0945944806 0.325029563 0.137097435 5.37765924
mDC          0.014860286 0.0237761816 0.030831149 0.008875553 5.03722483
B            0.003342805 0.0035032662 0.012989518 0.004005391 3.18020059
Plasma       0.015817544 0.0002244276 0.002149909 0.050912723 1.37935620
Macrophages  0.750869889 0.7352897831 0.512995844 0.604316237 1.21863016
Epithelial   0.053652014 0.1302459194 0.051903406 0.118455372 0.30529601
Mast         0.001063620 0.0023373349 0.002220485 0.001699001 0.04651485
                P.Value        FDR
pDC         0.007891726 0.02801024
Neutrophil  0.008006769 0.02801024
NK          0.008403072 0.02801024
T           0.023311489 0.05466535
mDC         0.027332677 0.05466535
B           0.080297449 0.13382908
Plasma      0.291776927 0.41544492
Macrophages 0.332355933 0.41544492
Epithelial  0.742906637 0.82545182
Mast        0.954732486 0.95473249</code></pre>
<pre class="r"><code>props.covid &lt;- getTransformedProps(clusters=covid$celltype, sample=covid$sample_new,
                                   transform=&quot;logit&quot;)</code></pre>
<pre class="r"><code>par(mfrow=c(1,1))
grp.covid &lt;- rep(c(&quot;Control&quot;,&quot;Moderate&quot;,&quot;Severe&quot;), c(4, 3, 6))
stripchart(as.numeric(props.covid$Proportions[&quot;Neutrophil&quot;,])~grp.covid,
           vertical=TRUE, pch=16, method=&quot;jitter&quot;,
           col = c(4,&quot;purple&quot;,2),cex=2, 
           ylab=&quot;Proportions&quot;,cex.axis=1.25, cex.lab=1.5)
title(&quot;Neutrophils in severe covid&quot;, cex.main=1.5, adj=0)
text(1.5,0.195, labels = &quot;Adj.Pval = 3.5e-06&quot;)</code></pre>
<p><img src="figure/RealDataAnalysis.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Number of samples in each group:</p>
<pre class="r"><code>table(grp.covid)</code></pre>
<pre><code>grp.covid
 Control Moderate   Severe 
       4        3        6 </code></pre>
<p>All significant cell types</p>
<pre class="r"><code>par(mfrow=c(2,2))
stripchart(as.numeric(props.covid$Proportions[&quot;Neutrophil&quot;,])~grp.covid,
           vertical=TRUE, pch=16, method=&quot;jitter&quot;,
           col = c(4,&quot;purple&quot;,2),cex=2, 
           ylab=&quot;Proportions&quot;,cex.axis=1.25, cex.lab=1.5)
title(&quot;Neutrophils in covid&quot;, cex.main=1.5, adj=0)

stripchart(as.numeric(props.covid$Proportions[&quot;Plasma&quot;,])~grp.covid,
           vertical=TRUE, pch=16, method=&quot;jitter&quot;,
           col = c(4,&quot;purple&quot;,2),cex=2, 
           ylab=&quot;Proportions&quot;,cex.axis=1.25, cex.lab=1.5)
title(&quot;Plasma in covid - outlier sample&quot;, cex.main=1.5, adj=0)

stripchart(as.numeric(props.covid$Proportions[&quot;pDC&quot;,])~grp.covid,
           vertical=TRUE, pch=16, method=&quot;jitter&quot;,
           col = c(4,&quot;purple&quot;,2),cex=2, 
           ylab=&quot;Proportions&quot;,cex.axis=1.25, cex.lab=1.5)
title(&quot;pDC in covid&quot;, cex.main=1.5, adj=0)

stripchart(as.numeric(props.covid$Proportions[&quot;NK&quot;,])~grp.covid,
           vertical=TRUE, pch=16, method=&quot;jitter&quot;,
           col = c(4,&quot;purple&quot;,2),cex=2, 
           ylab=&quot;Proportions&quot;,cex.axis=1.25, cex.lab=1.5)
title(&quot;NK in covid&quot;, cex.main=1.5, adj=0)</code></pre>
<p><img src="figure/RealDataAnalysis.Rmd/unnamed-chunk-26-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sig.covid &lt;- rownames(output.logit)

pdf(file=&quot;output/covidResults.pdf&quot;,width = 13, height=5)
par(mfrow=c(2,5))
par(mar=c(4,5,2,2))
for(i in 1:length(sig.covid)){
  
stripchart(as.numeric(props.covid$Proportions[sig.covid[i],])~grp.covid,
           vertical=TRUE, pch=16, method=&quot;jitter&quot;,
           col = c(4,&quot;purple&quot;,2),cex=2, 
           ylab=&quot;Proportions&quot;,cex.axis=1.25, cex.lab=1.5)

title(sig.covid[i], cex.main=1.5, adj=0)
legend(&quot;top&quot;, legend = paste(&quot;Adj.Pval = &quot;,round(output.asin[sig.covid[i],]$FDR,3),sep=&quot;&quot;),cex=1.5,bty=&quot;n&quot;,bg=&quot;n&quot;)

}
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
<div id="figure-3" class="section level1">
<h1>Figure 3</h1>
<pre class="r"><code>layout(matrix(c(1,1,2,3,3,4,5,5,6,7,7,7,8,8,8,9,9,9), 2, 9, byrow = TRUE))

par(mar=c(5.5,5.5,3,0))
barplot(prop.logit$Proportions, col=ggplotColors(nrow(prop.logit$Proportions)),
        cex.lab=1.5, cex.axis = 1.5, cex.names=1.5, ylab=&quot;Proportions&quot;,xlab=&quot;Samples&quot;,las=2)
title(&quot;a) Human heart development&quot;, adj=0,cex.main=1.5)
par(mar=c(0,0,0,0))
plot(1, type = &quot;n&quot;, xlab = &quot;&quot;, ylab = &quot;&quot;, xaxt=&quot;n&quot;,yaxt=&quot;n&quot;, bty=&quot;n&quot;)
legend(&quot;left&quot;,fill=ggplotColors(8),legend=rownames(prop.logit$Proportions), cex=1.25,
       bty=&quot;n&quot;)

par(mar=c(5.5,5.5,3,0))
par(mgp=c(4,1,0))
barplot(prop.list$Proportions, col=ggplotColors(24),las=2, cex.lab=1.5, 
        cex.axis = 1.5, cex.names=1.5, ylab=&quot;Proportions&quot;,xlab=&quot;Samples&quot;)
title(&quot;b) Old and young PBMCs&quot;, adj=0,cex.main=1.5)

par(mar=c(0,0,0,0))
plot(1, type = &quot;n&quot;, xlab = &quot;&quot;, ylab = &quot;&quot;, xaxt=&quot;n&quot;,yaxt=&quot;n&quot;, bty=&quot;n&quot;)
legend(&quot;left&quot;,fill=ggplotColors(24),legend=rownames(prop.list$Proportions), cex=1.25,
       bty=&quot;n&quot;)


par(mar=c(5.5,5.5,3,0))
barplot(props.covid$Proportions, col=ggplotColors(nrow(props.covid$Proportions)),
        cex.lab=1.5, cex.axis = 1.5, cex.names=1.5, ylab=&quot;Proportions&quot;,xlab=&quot;Samples&quot;,las=2)
title(&quot;c) COVID vs healthy controls &quot;, adj=0,cex.main=1.5)

par(mar=c(0,0,0,0))
plot(1, type = &quot;n&quot;, xlab = &quot;&quot;, ylab = &quot;&quot;, xaxt=&quot;n&quot;,yaxt=&quot;n&quot;, bty=&quot;n&quot;)
legend(&quot;left&quot;,fill=ggplotColors(10),legend=rownames(props.covid$Proportions), cex=1.25,
       bty=&quot;n&quot;)

par(mar=c(5,5.5,3,2))
stripchart(as.numeric(prop.logit$Proportions[&quot;Cardiomyocytes&quot;,])~grp,
           vertical=TRUE, pch=16, method=&quot;jitter&quot;,
           col = ggplotColors(4),cex=2, 
           ylab=&quot;Proportions&quot;,cex.axis=1.5, cex.lab=1.5)
title(&quot;d) Cardiomyocyte development&quot;, cex.main=1.5, adj=0)
abline(a=fit.plot$coefficients[&quot;Cardiomyocytes&quot;,1], b=fit.plot$coefficients[&quot;Cardiomyocytes&quot;,2], lty=2, lwd=2)
text(2.6,0.77, labels = &quot;Adj.Pval = 0.01&quot;,cex=1.5)


stripchart(as.numeric(sexprops[&quot;CD8.Naive&quot;,])~group.immune,
           vertical=TRUE, pch=c(8,16), method=&quot;jitter&quot;,
           col = c(ggplotColors(20)[20],&quot;hotpink&quot;,4, &quot;darkblue&quot;),cex=2,
           ylab=&quot;Proportions&quot;, cex.axis=1.5, cex.lab=1.5,
           group.names=c(&quot;F.Old&quot;,&quot;F.Young&quot;,&quot;M.Old&quot;,&quot;M.Young&quot;))
title(&quot;e) CD8.Naive: Young Vs Old&quot;, cex.main=1.5, adj=0)
text(3.2,0.18, labels = &quot;Adj.Pval = 0.004&quot;,cex=1.5)



grp.covid &lt;- rep(c(&quot;Control&quot;,&quot;Moderate&quot;,&quot;Severe&quot;), c(4, 3, 6))
stripchart(as.numeric(props.covid$Proportions[&quot;Neutrophil&quot;,])~grp.covid,
           vertical=TRUE, pch=16, method=&quot;jitter&quot;,
           col = c(4,&quot;purple&quot;,2),cex=2, 
           ylab=&quot;Proportions&quot;,cex.axis=1.5, cex.lab=1.5)
title(&quot;f) Neutrophils in severe covid&quot;, cex.main=1.5, adj=0)
text(1.5,0.195, labels = &quot;Adj.Pval = 0.028&quot;,cex=1.5)</code></pre>
<p><img src="figure/RealDataAnalysis.Rmd/unnamed-chunk-28-1.png" width="1440" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] gt_0.6.0        pheatmap_1.0.12 edgeR_3.38.1    limma_3.52.1   
[5] speckle_0.99.0  workflowr_1.7.0

loaded via a namespace (and not attached):
  [1] backports_1.4.1             plyr_1.8.7                 
  [3] igraph_1.3.1                lazyeval_0.2.2             
  [5] sp_1.4-7                    splines_4.2.0              
  [7] listenv_0.8.0               scattermore_0.8            
  [9] GenomeInfoDb_1.32.2         ggplot2_3.3.6              
 [11] digest_0.6.29               htmltools_0.5.2            
 [13] fansi_1.0.3                 checkmate_2.1.0            
 [15] magrittr_2.0.3              tensor_1.5                 
 [17] cluster_2.1.3               ROCR_1.0-11                
 [19] globals_0.15.0              matrixStats_0.62.0         
 [21] spatstat.sparse_2.1-1       colorspace_2.0-3           
 [23] ggrepel_0.9.1               xfun_0.31                  
 [25] dplyr_1.0.9                 callr_3.7.0                
 [27] crayon_1.5.1                RCurl_1.98-1.6             
 [29] jsonlite_1.8.0              progressr_0.10.0           
 [31] spatstat.data_2.2-0         survival_3.3-1             
 [33] zoo_1.8-10                  glue_1.6.2                 
 [35] polyclip_1.10-0             gtable_0.3.0               
 [37] zlibbioc_1.42.0             XVector_0.36.0             
 [39] leiden_0.4.2                DelayedArray_0.22.0        
 [41] future.apply_1.9.0          SingleCellExperiment_1.18.0
 [43] BiocGenerics_0.42.0         abind_1.4-5                
 [45] scales_1.2.0                DBI_1.1.2                  
 [47] spatstat.random_2.2-0       miniUI_0.1.1.1             
 [49] Rcpp_1.0.8.3                viridisLite_0.4.0          
 [51] xtable_1.8-4                reticulate_1.25            
 [53] spatstat.core_2.4-4         stats4_4.2.0               
 [55] htmlwidgets_1.5.4           httr_1.4.3                 
 [57] RColorBrewer_1.1-3          ellipsis_0.3.2             
 [59] Seurat_4.1.1                ica_1.0-2                  
 [61] pkgconfig_2.0.3             sass_0.4.1                 
 [63] uwot_0.1.11                 deldir_1.0-6               
 [65] locfit_1.5-9.5              utf8_1.2.2                 
 [67] tidyselect_1.1.2            rlang_1.0.2                
 [69] reshape2_1.4.4              later_1.3.0                
 [71] munsell_0.5.0               tools_4.2.0                
 [73] cli_3.3.0                   generics_0.1.2             
 [75] ggridges_0.5.3              evaluate_0.15              
 [77] stringr_1.4.0               fastmap_1.1.0              
 [79] yaml_2.3.5                  goftest_1.2-3              
 [81] processx_3.5.3              knitr_1.39                 
 [83] fs_1.5.2                    fitdistrplus_1.1-8         
 [85] purrr_0.3.4                 RANN_2.6.1                 
 [87] pbapply_1.5-0               future_1.26.1              
 [89] nlme_3.1-157                whisker_0.4                
 [91] mime_0.12                   compiler_4.2.0             
 [93] rstudioapi_0.13             plotly_4.10.0              
 [95] png_0.1-7                   spatstat.utils_2.3-1       
 [97] statmod_1.4.36              tibble_3.1.7               
 [99] bslib_0.3.1                 stringi_1.7.6              
[101] highr_0.9                   ps_1.7.0                   
[103] rgeos_0.5-9                 lattice_0.20-45            
[105] Matrix_1.4-1                vctrs_0.4.1                
[107] pillar_1.7.0                lifecycle_1.0.1            
[109] spatstat.geom_2.4-0         lmtest_0.9-40              
[111] jquerylib_0.1.4             RcppAnnoy_0.0.19           
[113] data.table_1.14.2           cowplot_1.1.1              
[115] bitops_1.0-7                irlba_2.3.5                
[117] GenomicRanges_1.48.0        httpuv_1.6.5               
[119] patchwork_1.1.1             R6_2.5.1                   
[121] promises_1.2.0.1            KernSmooth_2.23-20         
[123] gridExtra_2.3               IRanges_2.30.0             
[125] parallelly_1.31.1           codetools_0.2-18           
[127] MASS_7.3-57                 assertthat_0.2.1           
[129] SummarizedExperiment_1.26.1 rprojroot_2.0.3            
[131] SeuratObject_4.1.0          sctransform_0.3.3          
[133] S4Vectors_0.34.0            GenomeInfoDbData_1.2.8     
[135] mgcv_1.8-40                 parallel_4.2.0             
[137] grid_4.2.0                  rpart_4.1.16               
[139] tidyr_1.2.0                 rmarkdown_2.14             
[141] MatrixGenerics_1.8.0        Rtsne_0.16                 
[143] git2r_0.30.1                getPass_0.2-2              
[145] Biobase_2.56.0              shiny_1.7.1                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
